src/adapter/ember/adapter/emberAdapter.ts

```diff
@@ -681,9 +676,13 @@ export class EmberAdapter extends Adapter {
             gpdHeader.writeUInt8(gpdCommandId, 13);// commandID
             gpdHeader.writeUInt8(gpdCommandPayload.length, 14);// payloadSize
 
-            const gpFrame = ZclFrame.fromBuffer(Cluster.greenPower.ID, Buffer.concat([gpdHeader, gpdCommandPayload]));
+            const data = Buffer.concat([gpdHeader, gpdCommandPayload]);
+            const header = ZclHeader.fromBuffer(data);
+            const gpFrame = ZclFrame.fromBuffer(Cluster.greenPower.ID, header, data);
```

Comment:
@Nerivec this is the only place where I couldn't remove `ZclFrame.fromBuffer`, it's not really a problem since this cluster is always greenPower here (very unlikely that we will require custom cluster definitions for greenPower), but still. Do you see a way to do ` wasBroadcast: (gpFrame.Payload.gppNwkAddr != null) ? false : true,` without parsing the frame?

Reply:
According to [this](https://github.com/Koenkk/zigbee-herdsman/blob/cd6906cfc37d74a00e7f7bd7c1745f566441dda8/src/controller/greenPower.ts#L76), it should be fine to set `wasBroadcast: true`.